windows-release/msi-steps.yml

```diff
@@ -114,3 +114,26 @@ steps:
   - publish: '$(Build.ArtifactStagingDirectory)\msi'
     artifact: msi
     displayName: 'Publish MSI'
+
+  - powershell: >
+      $(Python)
+      "$(Build.SourcesDirectory)\sbom.py"
+      "--cpython-source-dir=$(Build.SourcesDirectory)"
+      $(gci -r "$(Build.ArtifactStagingDirectory)\msi\**\python-*.exe")
+    workingDirectory: $(Build.BinariesDirectory)
+    displayName: 'Create SBOMs for binaries'
+
+  - task: CopyFiles@2
+    displayName: 'Layout Artifact: sbom'
+    inputs:
+      sourceFolder: $(Build.ArtifactStagingDirectory)\msi
+      targetFolder: $(Build.ArtifactStagingDirectory)\sbom
+      flatten: true
+      contents: |
+        **\*.spdx.json
```

Comment:
Would be nice to put the above two steps before the previous `publish` step. It's uncommon, but when a task fails for no apparent reason, we can usually retry the job. However, if anything has been published, the publish step will fail and we need to restart the entire build. So ideally, we want to do all the work first and then publish.

I don't have any concerns about including the SBOMs in the msi artifact.

Reply:
Done in https://github.com/python/release-tools/pull/100/commits/c01765dacd7876de68e712362e2f8015aafae02b, also moved the positioning in the other job so publishing is last.